src/screens/SettingsScreen.js

```diff
@@ -18,11 +25,111 @@ import {
 } from '../components';
 import { SettingsListItem } from '../components/SettingsListItem';
 import { arrowLeft } from '../icons';
+import { createMatomoUserId, removeMatomoUserId, storageHelper } from '../helpers';
+import { useMatomoTrackScreenView } from '../hooks';
+
+const { MATOMO_TRACKING } = consts;
 
 export const SettingsScreen = ({ navigation }) => {
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const { orientation, dimensions } = useContext(OrientationContext);
   const [refreshing, setRefreshing] = useState(false);
+  const [sectionedData, setSectionedData] = useState([]);
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.SETTINGS);
+
+  useEffect(() => {
+    const updateSectionedData = async () => {
+      const { consent: matomoValue } = await storageHelper.matomoSettings();
+
+      setSectionedData([
+        {
+          data: [
+            {
+              title: texts.settingsTitles.pushNotifications,
+              topDivider: true,
+              type: 'toggle',
+              value: false
+            }
+          ]
+        },
+        {
+          data: [
+            {
+              title: texts.settingsTitles.analytics,
+              topDivider: true,
+              type: 'toggle',
+              value: matomoValue,
+              onActivate: (revert) =>
+                setTimeout(() => {
+                  Alert.alert(
+                    texts.settingsTitles.analytics,
+                    texts.settingsContents.analytics.onActivate,
+                    [
+                      {
+                        text: 'Nein',
+                        onPress: () => revert(),
+                        style: 'cancel'
+                      },
+                      { text: 'Ja', onPress: createMatomoUserId }
+                    ],
+                    { cancelable: false }
+                  );
+                }, 300),
```

Comment:
used timeout here to wait for the toggle animation in order to not have visual side effects

Reply:
I don't feel it is necessary. does it behave weirdly on android?

Reply:
i had the effect, that when toggling per switch click on iOS, it was not as smooth as expected sometimes, because the animation and alert are simultaneously.
on the other hand users wouldn't see the activity indicator. but maybe this is not necessary as the alert comes in.

Reply:
removed in https://github.com/ikuseiGmbH/smart-village-app-app/pull/125/commits/662a758d48bc2bf7531d482b634f25ff62fbaeb2